src/main/java/eventprocessing/storage/MessageLog.java

```diff
@@ -0,0 +1,26 @@
+package eventprocessing.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessageLog {
+
+    private List<String> messageHistory = new ArrayList<>();
+    private int maxSize;

```

Comment:
If you make both of these fields final, it makes it clear they're never reassigned.